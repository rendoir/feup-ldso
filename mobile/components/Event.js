import React from 'react';
import {
    StyleSheet,
    Image
} from 'react-native';

import { Card, Icon, View, Badge, Text } from 'native-base';

export default class Event extends React.Component {

    constructor(props) {
        super(props);

        this.state = {
            imageLoaded: true
        };

        this.onFavoriteCall = this.onFavoriteCall.bind(this);

    }

    ImageLoadingError() {
        this.setState({ imageLoaded: false });
    }

    getMonthInString(month) {
        return global.calendar[parseInt(month)][this.props.language];
    }

    onFavoriteCall() {
        this.props.onFavorite(this.props.id);
    }

    render() {
        return (
            <Card onPress={this.props.onPress} style={styles.card}>
                <Image source={this.state.imageLoaded ? { uri: 'http://' + global.api + ':3030/' + this.props.id } : require('../assets/images/default.png')}
                    style={styles.image}
                    onError={this.ImageLoadingError.bind(this)}
                    onPress={this.props.onPress} />
                <View style={styles.badgeView} onPress={this.props.onPress}>
                    <Badge style={styles.badge}>
                        <Text style={styles.badgeText} onPress={this.props.onPress}>{this.props.start_date.split('T')[0].split('-')[2]}</Text>
                        <Text style={styles.badgeText} onPress={this.props.onPress}>{this.getMonthInString(this.props.start_date.split('T')[0].split('-')[1])}</Text>
                    </Badge>
                    <View style={styles.mainView} onPress={this.props.onPress}>
                        <Text style={styles.title} numberOfLines={1} onPress={this.props.onPress}>{this.getTitle()}</Text>
                        <Text style={styles.simpleText} numberOfLines={1} onPress={this.props.onPress}>{this.props.location} - {this.props.start_date.split('T')[1].split(':')[0] + ':' + this.props.start_date.split('T')[1].split(':')[1]}</Text>
                        <Text style={styles.simpleText} numberOfLines={1} onPress={this.props.onPress}>{global.dictionary["PRICE"][this.props.language]}: {this.getPrice()}</Text>
                    </View>
                    <Icon className="fave_icon" style={styles.faveIcon} type='FontAwesome' name={this.props.is_favorite ? 'heart' : 'heart-o'} onPress={this.onFavoriteCall} />
                </View>
            </Card>
        );
    }

    getTitle() {
        if (this.props.language === "PT")      return this.props.title;
        else if (this.props.language === "EN") return this.props.title_english;
    }

    getPrice() {
        return (this.props.price == 0) ? global.dictionary["FREE"][this.props.language] : (this.props.price + 'â‚¬');
    }
}

const styles = StyleSheet.create({
    image: {
        height: 160,
        width: null,
        flex: 1
    },
    faveIcon: {
        fontSize: 35,
        flex: 2,
        alignSelf: 'center',
        color: '#FF5722'
    },
    simpleText: {
        color: 'black',
        fontFamily: 'OpenSans-Regular',
        fontSize: 16
    },
    title: {
        color: 'black',
        fontFamily: 'OpenSans-Regular',
        fontSize: 18,
        fontWeight: 'bold'
    },
    badgeText: {
        color: 'white',
        fontFamily: 'OpenSans-Regular',
        fontSize: 18,
        lineHeight: 30
    },
    badge: {
        flex: 2,
        backgroundColor: '#455A64',
        margin: '3%',
        height: 68,
        alignItems: 'center'
    },
    badgeView: {
        flexDirection: 'row',
        backgroundColor: 'white'
    },
    mainView: {
        flex: 9,
        marginRight: '3%',
        marginVertical: '3%'
    },
    card: {
        backgroundColor: 'white'
    }
});
