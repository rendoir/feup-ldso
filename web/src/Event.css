div.event {
    margin: 1em 0;
}

div.event-date {
    background-color: #4a4a4a;
    color: white;
    font-weight: bold;
    padding: 1em;
}

div.event-info {
    background-color: white;
    padding: 1em;
    text-align: left;
    position: relative;
    font-size: 18px;
}

div.event-info p.event-title {
    cursor: pointer;
    display: inline-block;
}

div.event-info p.event-title:hover {
    color: #4a4a4a;
}

div.event-info div.event-buttons{
    position: absolute;
    bottom: 8px;
    right: 8px;
}

div.event-info div.event-buttons button.btn{
    background-color: inherit;
    border-color: white;
    color: #4a4a4a;
}