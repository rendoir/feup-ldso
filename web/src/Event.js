import React, { Component } from 'react';
import { Col, Button, Row } from 'react-bootstrap';
import { Link } from 'react-router-dom';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import axios from 'axios';
import swal from 'sweetalert';
import './Event.css';

const monthNames = ["JANEIRO", "FEBREIRO", "MARÃ‡O", "ABRIL", "MAIO", "JUNHO",
    "JULHO", "AGOSTO", "SETEMBRO", "OUTUBRO", "NOVEMBRO", "DEZEMBRO"
];

function getTokenFromCookie() {
    let token = document.cookie.split("access_token=")[1];
    return token;
}

class Event extends Component {

    constructor(props) {
        super(props);

        var date = new Date(props.info.start_date);

        this.state = {
            day: (date.getDate() < 10 ? '0' : '') + date.getDate(),
            month: monthNames[date.getMonth()],
            year: date.getFullYear()
        };
        this.deleteEvent = this.deleteEvent.bind(this);
        this.deleteEventRequest = this.deleteEventRequest.bind(this);
    }

    deleteEvent() {
        swal({
            title: "Tem a certeza que quer apagar o evento?",
            icon: "warning",
            buttons: true,
            dangerMode: true
        })
            .then((isConfirm) => {
                if (isConfirm)
                    this.deleteEventRequest();

            });
    }

    deleteEventRequest() {
        let self = this;
        axios.delete('http://' + process.env.REACT_APP_API_URL + ':3030/', {
            data: {
                user_id: 1,
                id: self.props.info.id
            },
            headers: {'Authorization': "Bearer " + getTokenFromCookie()}
        }).then(() => self.props.deleteEventFromArray(self.props.info.id))
            .catch((err) => self.props.updateAlertMessage('danger', 'Ocorreu um erro a apagar o evento.:' + err));
    }

    render() {
        return (
            <Row className="event">
                <Col sm={2} className="event-date">
                    <h4>{this.state.day}</h4>
                    <h4>{this.state.month}</h4>
                    <h4>{this.state.year}</h4>
                </Col>
                <Col sm={10} className="event-info">
                    <Link to={`/event/${this.props.info.id}`}>
                        <p className="event-title" onClick={this.props.showEventPage}>{this.props.info.title}</p>
                    </Link>
                    <p>{this.props.info.initials}</p>

                    <div className="event-buttons">
                        <Button><FontAwesomeIcon icon="edit" /></Button>
                        <Button onClick={this.deleteEvent} className="delete-button"><FontAwesomeIcon icon="trash-alt" /></Button>
                    </div>
                </Col>
            </Row>
        );
    }
}

export default Event;
