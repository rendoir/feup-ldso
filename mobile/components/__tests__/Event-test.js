/**
 * @jest-environment jsdom
 */

import 'react-native';
import React from 'react';
import Event from '../Event';
import renderer from 'react-test-renderer';


it('renders correctly', () => {
    eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "poster_id": 2 };
    const tree = renderer.create(<Event data={eventData} />).toJSON();

    expect(tree).toMatchSnapshot();
});

it('getMonthInString returns correctly', () => {
    eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "poster_id": 2 };
    const tree = renderer.create(<Event data={eventData} />).getInstance();

    expect(tree.getMonthInString('01')).toEqual("JAN");
    expect(tree.getMonthInString('02')).toEqual("FEV");
    expect(tree.getMonthInString('03')).toEqual("MAR");
    expect(tree.getMonthInString('04')).toEqual("ABR");
    expect(tree.getMonthInString('05')).toEqual("MAI");
    expect(tree.getMonthInString('06')).toEqual("JUN");
    expect(tree.getMonthInString('07')).toEqual("JUL");
    expect(tree.getMonthInString('08')).toEqual("AGO");
    expect(tree.getMonthInString('09')).toEqual("SET");
    expect(tree.getMonthInString('10')).toEqual("OUT");
    expect(tree.getMonthInString('11')).toEqual("NOV");
    expect(tree.getMonthInString('12')).toEqual("DEZ");
    expect(tree.getMonthInString('asjdnawd')).toEqual("Error");
});

it('image loading', () => {
    eventData = { "id": 2, "title": "Feup Career Fair 2018", "description": "Nos dias 16, 17, 18 e 19 de outubro acontece mais uma edição da FEUP CAREER FAIR e a FEUP convida todos os estudantes e graduados a estarem presentes!", "start_date": "2018-10-15T23:00:00.000Z", "end_date": "2018-10-18T23:00:00.000Z", "location": "FEUP", "price": 0, "entity_id": 9, "poster_id": 2 };
    const tree = renderer.create(<Event data={eventData} />).getInstance();

    tree.ImageLoadingError();
    expect(tree.state.imageLoaded).toEqual(false);
});