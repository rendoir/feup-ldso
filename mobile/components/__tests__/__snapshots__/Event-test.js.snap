// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event Test renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "http://188.166.95.23:3030/2",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 160,
        "width": null,
      }
    }
  />
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "white",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#ED1727",
            "borderRadius": 13.5,
            "height": 27,
            "justifyContent": "center",
            "paddingBottom": 3,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#455A64",
            "flex": 1,
            "height": 68,
            "margin": "3%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 15,
              "lineHeight": 19,
              "paddingLeft": 3,
              "paddingRight": 3,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "lineHeight": 30,
            },
          ]
        }
        uppercase={false}
      >
        15
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 15,
              "lineHeight": 19,
              "paddingLeft": 3,
              "paddingRight": 3,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "lineHeight": 30,
            },
          ]
        }
        uppercase={false}
      >
        OUT
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 5,
            "marginRight": "3%",
            "marginVertical": "3%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Feup Career Fair 2018
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
        uppercase={false}
      >
        FEUP
         - 
        23:00
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
        uppercase={false}
      >
        Pre√ßo
        : 
        Gratuito
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;

exports[`Event Test renders correctly in english 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#fff",
        "borderBottomWidth": 0.5,
        "borderColor": "#ccc",
        "borderLeftWidth": 0.5,
        "borderRadius": 2,
        "borderRightWidth": 0.5,
        "borderTopWidth": 0.5,
        "elevation": 3,
        "flexWrap": "nowrap",
        "marginBottom": 5,
        "marginLeft": 2,
        "marginRight": 2,
        "marginTop": 5,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.1,
        "shadowRadius": 1.5,
      },
      Object {
        "backgroundColor": "white",
      },
    ]
  }
>
  <Image
    onError={[Function]}
    source={
      Object {
        "uri": "http://188.166.95.23:3030/2",
      }
    }
    style={
      Object {
        "flex": 1,
        "height": 160,
        "width": null,
      }
    }
  />
  <View
    style={
      Array [
        Object {},
        Object {
          "backgroundColor": "white",
          "flexDirection": "row",
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "alignSelf": "flex-start",
            "backgroundColor": "#ED1727",
            "borderRadius": 13.5,
            "height": 27,
            "justifyContent": "center",
            "paddingBottom": 3,
            "paddingLeft": 6,
            "paddingRight": 6,
            "paddingTop": 3,
          },
          Object {
            "alignItems": "center",
            "backgroundColor": "#455A64",
            "flex": 1,
            "height": 68,
            "margin": "3%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 15,
              "lineHeight": 19,
              "paddingLeft": 3,
              "paddingRight": 3,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "lineHeight": 30,
            },
          ]
        }
        uppercase={false}
      >
        15
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": "#fff",
              "fontFamily": "System",
              "fontSize": 15,
              "lineHeight": 19,
              "paddingLeft": 3,
              "paddingRight": 3,
              "textAlign": "center",
            },
            Object {
              "color": "white",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "lineHeight": 30,
            },
          ]
        }
        uppercase={false}
      >
        OCT
      </Text>
    </View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 5,
            "marginRight": "3%",
            "marginVertical": "3%",
          },
        ]
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 18,
              "fontWeight": "bold",
            },
          ]
        }
        uppercase={false}
      >
        Feup Career Fair 2018
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
        uppercase={false}
      >
        FEUP
         - 
        23:00
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#000",
              "fontFamily": "System",
              "fontSize": 16,
            },
            Object {
              "color": "black",
              "fontFamily": "OpenSans-Regular",
              "fontSize": 16,
            },
          ]
        }
        uppercase={false}
      >
        Price
        : 
        Free
      </Text>
    </View>
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
    />
  </View>
</View>
`;
